lib/github.js

```diff
@@ -10,51 +10,32 @@ const GitHubApi = require('github')
  * @see {@link https://github.com/octokit/node-github}
  */
 
-// Default callback should just return the response passed to it.
 const defaultCallback = response => response
-
-class EnhancedGitHubClient extends GitHubApi {
-  constructor (options) {
-    super(options)
-    this.limiter = new Bottleneck(1, 1000)
-    this.logger = options.logger
+async function paginate (octokit, responsePromise, callback = defaultCallback) {
+  let collection = []
+  let getNextPage = true
+  let done = () => {
+    getNextPage = false
   }
+  let response = await responsePromise
+  collection = collection.concat(await callback(response, done))
+ // eslint-disable-next-line no-unmodified-loop-condition
+  while (getNextPage && octokit.hasNextPage(response)) {
+    response = await octokit.getNextPage(response)
+    collection = collection.concat(await callback(response, done))
+  }
+  return collection
+}
 
-  handler (params, block, callback) {
-    // Only allow one request at a time with a 1s delay
-    // https://github.com/octokit/node-github/issues/526
-    this.limiter.submit(super.handler.bind(this), params, block, (err, res) => {
-      let msg = `GitHub request: ${block.method} ${block.url}`
-      if (res) {
-        msg += ` - ${res.meta.status}`
```

Comment:
did this ever work? I can’t recall that `github` returned a `status` property on `meta`

Reply:
nevermind, `res.meta` is actually the headers from the response, I just had it changed in `@octokit/rest` to `res.headers`. I changed it back to keep full compatibility. Nice catch there probot 🙏